clientapi/auth/login.go

```diff
@@ -32,8 +31,13 @@ import (
 // called after authorization has completed, with the result of the authorization.
 // If the final return value is non-nil, an error occurred and the cleanup function
 // is nil.
-func LoginFromJSONReader(ctx context.Context, r io.Reader, useraccountAPI uapi.UserLoginAPI, userAPI UserInternalAPIForLogin, cfg *config.ClientAPI) (*Login, LoginCleanupFunc, *util.JSONResponse) {
-	reqBytes, err := io.ReadAll(r)
+func LoginFromJSONReader(
```

Comment:
I think we should keep the current function signature, as having to pass in a `*http.Request` makes testing harder.

Reply:
...this on the other hand, I cannot really get behind. I do get that we would like to keep the current signature, but based on how the application server works, we also need to check the header information regarding the tokens, so we would need to have it based on an actual (mock up) http request, at least that's the sentiment I'm getting from the code.
Would it help if we split that into a 2nd login method, just for AS-based logins?

Reply:
I do, however, agree on the sentiment of not changing existing function signatures, so, this would actually be something a "functional designer" would need to make a decision on, if it is still used somewhere else or cause additional issues. From the state I saw it / tried it, it works as expected, but, again, I do agree that there may be a better solution to do this, I just can't really think of any in the context of the Bearer token check in this particular case.